src/seedu/addressbook/data/AddressBook.java

```diff
@@ -22,13 +25,14 @@
 
     private final UniquePersonList allPersons;
     private final UniqueTagList allTags; // can contain tags not attached to any person
-
+    private final List<Tagging> allTaggings;
```

Comment:
Correct, this implements the class diagram properly. Any reason why you decided to make it final as well?